examples/chardev2.c

```diff
@@ -162,13 +159,21 @@ device_ioctl(struct file *file, /* ditto */
         put_user('\0', (char __user *)ioctl_param + i);
         break;
     }
-    case IOCTL_GET_NTH_BYTE:
+    case IOCTL_GET_NTH_BYTE: {
         /* This ioctl is both input (ioctl_param) and output (the return
          * value of this function).
          */
-        return (long)message[ioctl_param];
-        break;
+        long ret = (long)message[ioctl_param];
```

Comment:
Move the declaration or `ret` before `switch` statement, so that we can avoid the repeated statement `atomic_set(&already_open, CDEV_NOT_USED)`.